### PR TITLE
Move ImmutableOpenMap usage to j.u.Map and upgrade gradle to 7.6.1

### PR DESCRIPTION
**Pull request by jmazanec15**
### Description

Core deleted ImmutableOpenMap in https://github.com/opensearch-project/OpenSearch/pull/7309. PR removes usages of this class. Additionally, build is failing due to gradle 7.5 issue - PR changes gradle to 7.6.1 as was done in #878. Lastly, updating BWC tests for SNAPSHOT-2.8.
 
Gradle was updated with following command:
```
gradle wrapper --gradle-version 7.6.1
```

### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by jmazanec15** [comment]
Secure test is failing:
```
java.lang.NoSuchMethodError: 'java.util.Map org.opensearch.cluster.node.DiscoveryNodes.getDataNodes()'
        at org.opensearch.knn.plugin.transport.TrainingJobRouterTransportAction.selectNode(TrainingJobRouterTransportAction.java:99)
        at org.opensearch.knn.plugin.transport.TrainingJobRouterTransportAction.lambda$routeRequest$1(TrainingJobRouterTransportAction.java:75)
        at org.opensearch.action.ActionListener$1.onResponse(ActionListener.java:80)
        at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:113)
        at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:107)
        at org.opensearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:73)
        at org.opensearch.action.ActionRunnable$2.doRun(ActionRunnable.java:88)
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
        at org.opensearch.common.util.concurrent.OpenSearchExecutors$DirectExecutorService.execute(OpenSearchExecutors.java:341)
        at org.opensearch.action.support.nodes.TransportNodesAction$AsyncAction.finishHim(TransportNodesAction.java:315)
        at org.opensearch.action.support.nodes.TransportNodesAction$AsyncAction.onOperation(TransportNodesAction.java:300)
        at org.opensearch.action.support.nodes.TransportNodesAction$AsyncAction$1.handleResponse(TransportNodesAction.java:277)
        at org.opensearch.action.support.nodes.TransportNodesAction$AsyncAction$1.handleResponse(TransportNodesAction.java:269)
        at org.opensearch.transport.TransportService$6.handleResponse(TransportService.java:777)
        at org.opensearch.security.transport.SecurityInterceptor$RestoringTransportResponseHandler.handleResponse(SecurityInterceptor.java:306)
        at org.opensearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1381)
        at org.opensearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1464)
        at org.opensearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1444)
        at org.opensearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:70)
        at org.opensearch.action.support.nodes.TransportNodesAction$NodeTransportHandler.messageReceived(TransportNodesAction.java:328)
        at org.opensearch.security.ssl.transport.SecuritySSLRequestHandler.messageReceivedDecorate(SecuritySSLRequestHandler.java:192)
        at org.opensearch.security.transport.SecurityRequestHandler.messageReceivedDecorate(SecurityRequestHandler.java:170)
        at org.opensearch.security.ssl.transport.SecuritySSLRequestHandler.messageReceived(SecuritySSLRequestHandler.java:96)
        at org.opensearch.security.OpenSearchSecurityPlugin$7$1.messageReceived(OpenSearchSecurityPlugin.java:675)
        at org.opensearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:106)
        at org.opensearch.transport.TransportService$8.doRun(TransportService.java:973)
        at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:806)
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1589)
```

This is because 3.0.0 staging docker image is old and doesnt incorporate the ImmutableOpenMap change: https://hub.docker.com/layers/opensearchstaging/opensearch/3.0.0/images/sha256-3f3dbc965fcaf8627a4b35996e57548aedf346083fb220484878f93412759e11?context=explore

File: src/main/java/org/opensearch/knn/plugin/transport/TrainingJobRouterTransportAction.java [hunk 1]
```diff
@@ -21,13 +21,15 @@
 import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.common.Strings;
 import org.opensearch.common.ValidationException;
-import org.opensearch.common.collect.ImmutableOpenMap;
+// import org.opensearch.common.collect.ImmutableOpenMap;
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/knn/plugin/transport/TrainingJobRouterTransportAction.java:24]: seems this line is not needed

Comment (jmazanec15) [reply src/main/java/org/opensearch/knn/plugin/transport/TrainingJobRouterTransportAction.java:24]: good catch

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,12 +1,6 @@
-#
-#  Copyright OpenSearch Contributors
-#  SPDX-License-Identifier: Apache-2.0
-#
-
```
Comment (martin-gaievski) [inline gradle/wrapper/gradle-wrapper.properties:5]: I think we still need to have a license header, please check core repo, they kept it after gradle 8.1.1 upgrade https://github.com/opensearch-project/OpenSearch/blob/main/gradle/wrapper/gradle-wrapper.properties

Comment (jmazanec15) [reply gradle/wrapper/gradle-wrapper.properties:5]: Good catch

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,12 +1,6 @@
...
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
+networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionSha256Sum=97a52d145762adc241bad7fd18289bf7f6801e08ece6badf80402fe2b9f250b1
```
Comment (heemin32) [inline gradle/wrapper/gradle-wrapper.properties:11]: can we have sha256 for new version of gradle?

Comment (jmazanec15) [reply gradle/wrapper/gradle-wrapper.properties:11]: Good point. Updated.

**Review by martin-gaievski — APPROVED** [review]

**Review by heemin32 — APPROVED** [review]

**Comment by nknize** [comment]
[`ImmutableOpenMap` removal backport PR landed](https://github.com/opensearch-project/OpenSearch/pull/8316) so this needs to be backported.

**Comment by navneet1v** [comment]
@junqiu-lei the release owner for 2.9. can we backport this PR to 2.x

**Pull request merged**